src/ui/components/Library/Team/View/TestRuns/suspense/TestsCache.ts

```diff
@@ -10,19 +10,29 @@ import { Test } from "shared/test-suites/TestRun";
 
 import { GET_WORKSPACE_TESTS } from "../../Tests/graphql/TestGraphQL";
 
+// We don't have any good way like (Test.created_at) to sort tests, therefore we can't properly build query ranges,
+// therefore can't use createIntervalCache
```

Comment:
Is this true? 😮  How is the backend filtering tests for the "last ___" filters then?

Don't we just need to add that attribute to the `GET_WORKSPACE_TESTS` query so that it gets returned to the client?

Reply:
Backend is building the list of test by grouping them together from list of test executions. Executions have created_at but I'm not sure how we can re-use that for tests. @ryanjduffy Any thoughts here?

Reply:
@callingmedic911 is correct that there isn't a timestamp for a test because it's a logical entity derived from the executions of it. In other words, the test exists in the result set because it was executed within the time frame specified. It's more like a category than an entity.

The closest thing we could do is echo back the `startTime` in some value. I'm not sure that's better than adapting the results on the client to return that value.

Reply:
> I'm not sure that's better than adapting the results on the client to return that value.

It would only be "better" in that it could result in a more efficient data fetching strategy on the frontend, but I guess it's not necessary.

Reply:
Can we adapt the data the is returned from graphql such that the FE could still fetch efficiently? I'd like to avoid adding a synthetic data element to the graphql response but we can do that if it is the only option.

Reply:
If we can't sort the data (client side) by date, then we can't use an interval cache to fetch and stitch together chunks of data, which means we'll need to re-fetch the list each time the date range changes. It's probably not _that bad_ but it does seem unneccessary.

FWIW I'm pretty far removed from how this stuff is actually architected at this point (bc I'm trying to just review code changes and otherwise focus on other stuff)